compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/pass/compiletime/reshaping/SamplingTaskSizingPass.java

```diff
@@ -0,0 +1,468 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.compiler.optimizer.pass.compiletime.reshaping;
+
+import org.apache.nemo.common.Util;
+import org.apache.nemo.common.dag.Edge;
+import org.apache.nemo.common.ir.IRDAG;
+import org.apache.nemo.common.ir.edge.IREdge;
+import org.apache.nemo.common.ir.edge.executionproperty.*;
+import org.apache.nemo.common.ir.vertex.IRVertex;
+import org.apache.nemo.common.ir.vertex.executionproperty.EnableDynamicTaskSizingProperty;
+import org.apache.nemo.common.ir.vertex.executionproperty.ParallelismProperty;
+import org.apache.nemo.common.ir.vertex.utility.TaskSizeSplitterVertex;
+import org.apache.nemo.common.ir.vertex.utility.runtimepasstriggervertex.SignalVertex;
+import org.apache.nemo.compiler.optimizer.pass.compiletime.annotating.Annotates;
+import org.apache.nemo.runtime.common.plan.StagePartitioner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Compiler pass for dynamic task size optimization. Happens only when the edge property is SHUFFLE.
+ * If (size of given job) >= 1GB: enable dynamic task sizing optimization.
+ * else:                          break.
+ *
+ *
+ * @Attributes
+ * PARTITIONER_PROPERTY_FOR_SMALL_JOB:  PartitionerProperty for jobs in range of [1GB, 10GB) size.
+ * PARTITIONER_PROPERTY_FOR_MEDIUM_JOB: PartitionerProperty for jobs in range of [10GB, 100GB) size.
+ * PARTITIONER_PROPERTY_FOR_BIG_JOB:    PartitionerProperty for jobs in range of [100GB, - ) size(No upper limit).
+ *
+ * source stage - shuffle edge - current stage - next stage
+ * -> source stage - [curr stage - signal vertex] - next stage
+ * where [] is a splitter vertex
+ */
+@Annotates({EnableDynamicTaskSizingProperty.class, PartitionerProperty.class, SubPartitionSetProperty.class,
+  ParallelismProperty.class})
+public final class SamplingTaskSizingPass extends ReshapingPass {
+  private static final Logger LOG = LoggerFactory.getLogger(SamplingTaskSizingPass.class.getName());
+
+  private static final int PARTITIONER_PROPERTY_FOR_SMALL_JOB = 1024;
+  private static final int PARTITIONER_PROPERTY_FOR_MEDIUM_JOB = 2048;
+  private static final int PARTITIONER_PROPERTY_FOR_BIG_JOB = 4096;
+  private final StagePartitioner stagePartitioner = new StagePartitioner();
+
+  /**
+   * Default constructor.
+   */
+  public SamplingTaskSizingPass() {
+    super(SamplingTaskSizingPass.class);
+  }
+
+  @Override
+  public IRDAG apply(final IRDAG dag) {
+    /* Step 1. check DTS launch by job size */
+    boolean enableDynamicTaskSizing = getEnableFromJobSize(dag);
+    if (!enableDynamicTaskSizing) {
+      dag.topologicalDo(v -> {
+        v.setProperty(EnableDynamicTaskSizingProperty.of(enableDynamicTaskSizing));
+        for (IREdge e : dag.getIncomingEdgesOf(v)) { // do we need this? erase this code and check results
```

Comment:
Is this a todo?

Reply:
I found out that this is actually an unnecessary code, and I have erased it. Thanks.